tests-e2e/LocalOcw.ts

```diff
@@ -0,0 +1,199 @@
+import * as path from "node:path"
+import * as http from "node:http"
+import { SpawnOptions } from "node:child_process"
+import execShCb from "exec-sh"
+import handler from "serve-handler"
+import Table from "cli-table3"
+import * as color from "ansi-colors"
+
+import { TEST_SITES, LOCAL_OCW_PORT, siteUrl, TestSiteAlias } from "./util/test_sites"
+import SimpleServer, { RedirectionRule } from "./util/SimpleServer"
+import { env, cleanEnv, stringifyEnv } from "../env"
+import { hugo } from "../package_scripts/util"
+
+const execSh = execShCb.promise
+
+const fromRoot = (...pathFromRoot: string[]) => {
+  return path.join(__dirname, "../", ...pathFromRoot)
+}
+
+/**
+ * Redirect all requests /path/to/thing to /ocw-ci-test-www/path/to/thing,
+ * except requests to /courses/.
+ */
+const OCW_WWW_REDIRECT: RedirectionRule = {
+  type:      "rewrite",
+  match:     /^\/(?!(courses\/))/,
+  transform: url => `/ocw-ci-test-www${url}`
+}
+
+interface LocalOCWOptions {
+  /**
+   * All sites will be built to subdirectories of this directory.
+   */
+  rootDestinationDir: string
+  staticApiPort: number
+}
+
+class LocalOCW {
```

Comment:
Almost all of this code was previously in `global-setup.ts`, with one exception noted below. I moved it here for re-use in the jest tests. Aside from that, I think this is better organization.